sale_force_invoiced_quantity/models/sale_order_line.py

```diff
@@ -0,0 +1,30 @@
+# Copyright (C) 2024 Cetmix OÜ
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class SaleOrderLine(models.Model):
+    _inherit = "sale.order.line"
+
+    force_invoiced = fields.Float(
```

Comment:
I would have added a constraint to check the force_invoiced quantity is not greater than the ordered quantity (at least - better to take into account qty_invoiced too)

Reply:
There was such constraint however I [asked to remove](https://github.com/OCA/sale-workflow/pull/3233#discussion_r1677081703) it.
The reason is because in some cases there can be a negative quantity.

Example: you have a total contract for 500 memes with a customer. You have already delivered 100 memes to the customer and 400 remaining are still in progress.
Than you receive a feedback from your customer who says that your memes are not funny and he would like to cancel the order. During the negotiation you agree that you will supply 150 of memes for free.
So you will add "Force Invoiced" to -150 and when you will deliver remaining 400 memes amount to invoice will show 350. Which is what you need.

Reply:
Ok, let's see if current behavior causes problems or not.